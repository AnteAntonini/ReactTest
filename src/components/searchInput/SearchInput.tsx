import "./searchInput.css";
import { useEffect } from "react";

type TSearchInputParams = {
  placeholder?: string;
  message: string;
  onChange: any;
};

const SearchInput = ({
  placeholder,
  onChange,
  message,
}: TSearchInputParams) => {
  useEffect(() => {
    console.log(`${message} Search Input`);
  }, []);

  return (
    <div className="row">
      <input
        className="search"
        type="text"
        placeholder={placeholder ?? "Search..."}
        onChange={onChange}
        data-testid="search-input"
      />
    </div>
  );
};

export default SearchInput;
