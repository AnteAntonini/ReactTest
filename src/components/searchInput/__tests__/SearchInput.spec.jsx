import { render, fireEvent, screen } from "@testing-library/react";
import SearchInput from "../SearchInput";

describe("SearchInput Component", () => {
  it("renders the search input with a placeholder", () => {
    const placeholderText = "Enter your search query";

    render(<SearchInput placeholder={placeholderText} onChange={() => {}} />);

    const searchInput = screen.getByTestId("search-input");

    expect(searchInput).toBeInTheDocument();
    expect(searchInput).toHaveAttribute("placeholder", placeholderText);
  });

  it("calls onChange when the input value changes", () => {
    const mockOnChange = jest.fn();

    render(<SearchInput onChange={mockOnChange} />);

    const searchInput = screen.getByTestId("search-input");

    fireEvent.change(searchInput, { target: { value: "search query" } });

    expect(mockOnChange).toHaveBeenCalled();
  });
});

describe("Test snapshot", () => {
  it("Component should match snapshot", () => {
    const placeholderText = "Enter your search query";
    const { asFragment } = render(
      <SearchInput placeholder={placeholderText} onChange={() => {}} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
