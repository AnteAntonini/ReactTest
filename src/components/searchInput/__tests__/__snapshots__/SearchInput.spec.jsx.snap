// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <input
      class="search"
      data-testid="search-input"
      placeholder="Enter your search query"
      type="text"
    />
  </div>
</DocumentFragment>
`;
